### DISCUSSION TITLE
Allow or disallow ads on public instances

### DISCUSSION THREAD

**Discussion by unixfox** [searxng/searx-instances#253] [Ideas]
Currently, the rules allow ads to be displayed as long as no tracking is involved.

A new recently requested instance has started to display ads (without any notice): https://github.com/searxng/searx-instances/issues/251. Some team members were reluctant against this kind of thing: https://github.com/searxng/searx-instances/issues/251#issuecomment-1336356775

What I propose here is to discuss whenever public instances should be allowed to display ads, as part of for example generating some revenue for paying for the server that run the instance.

Should we allow ads on the public instances? 

✅ If you agree:
- What's the kind of type of ad that can be run? For example, would you allow displaying a banner at the top of the page?
<img src="https://user-images.githubusercontent.com/4016501/205492690-56c080e4-90fe-4cae-8094-3ea2a0c92fdf.png" width="300">

- Why should we allow ad to be run on the public instance?

Feel free to explain more than the questions.

❌ If you disagree:
- Why shouldn't we?
- What's considered an ad for you? Is displaying a banner about a product an ad?

Feel free to explain more than the questions.

-----

Pinging people that may want to react to this idea @return42 @dalf @tiekoetter @ononoki1 @mrpaulblack @stephenhodgkiss.

**Comment by stephenhodgkiss** [comment]
I agree, as long as: -

1. the pages are not bombarded with banners after every few search results for example. Top and bottom of any page is fine.
2. there is no analytical or tracking software installed on the instance itself.
3. having another clickable link that goes to a website that belongs to a third-party is expected. After all, the results are all hyperlinked to websites that the instance owner has zero control over. Those external websites may or may not having analytical or tracking on them which is not a concern here, otherwise nearly 99% of websites would be excluded from search results.

**Comment by TalOrenshtein** [comment]
I'm generally agree with @stephenhodgkiss . I just want to add that IMO banners are OK, but sponsored search results are not.

**Comment by unixfox** [comment]
https://old.reddit.com/r/Searx/comments/zd0ltg/we_need_your_opinion_on_whenever_public_instances/

Latest messages from Reddit:
<img width="874" alt="image" src="https://user-images.githubusercontent.com/4016501/205631395-50281317-a125-490a-a88a-53d79d549875.png">

**Comment by rasos** [comment]
I am in favor of a dedicated sponsors & advertisements box, which is positioned on all instances at the same UI area and should be labelled `Sponsored`. And in the [server list](https://searx.space/) we probably should also add a label `Ads` if a sponsored area has been activated. 

And yes, no ads which are controlled by user profiles and their preferences.

**Comment by catfluoride** [comment]
- I agree that instance with 1st party _non-personalized and non-tracking_ ads may be allowed.

- I agree with @rasos about adding the `Ads` label to server list.

- I don't know how feasible is to force the same UI are on all instances instances with Ads. This would be ideal, but it may soon become an additional burden for SearXNG (but I'm not a developer, so I can't tell how much effort is required for something like that). Perhaps a _suggestion_  may be made to instance admin to display ads on the right, besides search results.

**Reply by unixfox** [reply]
I think the ability to add customizable HTML code that is displayed at the very top of the instance is probably a better way of solving this issue. For example, Invidious already allow this kind of customization: https://github.com/iv-org/invidious/blob/master/config/config.example.yml#L425

About labeling, searx.space allow to add comments to a specific instance, maybe that could be a good candidate for that. We don't want to add too many columns in order to keep it "simple".

**Comment by fehho** [comment]
Static same-site advertisements should be fine so long as it's labeled as modified HTML and the listed response time includes downloading the extra content.

**Comment by ononoki1** [comment]
I don't think ads should be allowed on public instances. One of the reasons that I use SearXNG is to get rid of annoying ads from popular search engines (e.g. Google). And people may wonder whether the SearXNG community has something to do with those ads when they find such an instance on searx.space.

If we finally choose to allow public instances with ads, at least we should rank these instances much lower (not just labeled as modified HTML because ads is obviously much more annoying than that).

**Reply by ononoki1** [reply]
> If all instances were the same, what's the point of having an instance of your own?

Difference is good only when some instances are better than vanilla ones. Do you really think adding ads makes an instance better?

**Reply by stephenhodgkiss** [reply]
**One of the reasons that I use SearXNG is to get rid of annoying ads from popular search engines (e.g. Google)**

You can decide which of the 100 instances to choose from,

**Reply by stephenhodgkiss** [reply]
**If we finally choose to allow public instances with ads, at least we should rank these instances much lower (not just labeled as modified HTML because ads is obviously much more annoying than that).**

That would be ridiculous when they are just another hyperlink on a page ... especially when a developer spends so much time getting all the Grades up to A+.

**Reply by ononoki1** [reply]
> You can decide which of the 100 instances to choose from.

When we display instances on searx.space, we have responsibility to consider user experience. Since ads will annoy users, and there are over 100 instances without ads, why don't we just forbid instances with ads?

**Reply by ononoki1** [reply]
> That would be ridiculous when they are just another hyperlink on a page ... especially when a developer spends so much time getting all the Grades up to A+.

Ads are not just links. They take space on the page (and usually in a very eye-catching position). And instance maintainers should get some rewards by using a custom donation page instead of displaying ads.

**Comment by return42** [comment]
I fully agree with the words from @ononoki1 .. further **my very personal view on this topic**:

> especially when a developer spends so much time getting all the Grades up to A+.

@stephenhodgkiss Sorry but I never have seen a contribution to SearXNG from you, from this point you are not a developer.  Beside the copy you made from the SearXNG and where you did your modifications is not a git clone what it makes hard for other to track & compare the copy with SearXNG's master.  What I mean: the SearXNG contributors can't really generate a profit of the changes in your copy.

All I have seen are issues about your manipulation of SearXNG, manipulation that was made to place advertising.  And this is the point where it starts to bother me:  

SearXNG is open source, everyone can do with it what he wants (while having an eye on the licence), but no one can expect that the resources of SearXNG contributors are used for activities that deviate from the basic aspirations of the open source community and in specifics of the SearXNG community.

In this manner, I can only emphasize again what @ononoki1 has been said:

> One of the reasons that I use SearXNG is to get rid of annoying ads 

:+1:

**Comment by unixfox** [comment]
[old.reddit.com/r/Searx/comments/zd0ltg/we_need_your_opinion_on_whenever_public_instances](https://old.reddit.com/r/Searx/comments/zd0ltg/we_need_your_opinion_on_whenever_public_instances/)

Latest messages from Reddit:
![image](https://user-images.githubusercontent.com/4016501/205858643-b7808ee0-61b2-4346-8c89-bdd20962d53a.png)

**Reply by stephenhodgkiss** [reply]
Thank you, finally seem to have a sensible solution

**Comment by stephenhodgkiss** [comment]
@return42 
**All I have seen are issues about your manipulation of SearXNG, manipulation that was made to place advertising**

Utter crap and was not the case at all. The issue I raised was even before the banners were added and was specifically in relation to the HTML Grade column ... at that time there were no banners. For example it stated that the JavaScript seemed to have been changed but it hadn't been. You are just being offensive and again making assumptions and remarks that aren't true. Please refrain from such. 

For myself this is concluded and I will no longer respond to this thread. Thanks to @unixfox for taking the time to listen to the issue.

**Reply by return42** [reply]
I was talking about these issues

- https://github.com/searxng/searxng/issues/1994
- https://github.com/searxng/searxng/issues/2003 
  and the copy of the source tree linked in the issue 
   - https://github.com/stephenhodgkiss/searxng-master 

   ..  but no longer available.

**Reply by stephenhodgkiss** [reply]
and both unrelated to the banners !

GitHub repository removed as no longer a Public instance

**Reply by return42** [reply]
> and both unrelated to the banners !

My point was: If we support banners here on searx.space I expect more issues related manipulated HTML, CSS, JS .. at searxng/searxng (caused by the manipulations needed to place the advertisement) .. and that bothers me.

--> _.. but no one can expect that the resources of SearXNG contributors are used for activities that deviate from the basic aspirations of the open source community_

> GitHub repository removed as no longer a Public instance

By using SearXNG you accepted the [AGPL-3.0 license](https://github.com/searxng/searxng/blob/master/LICENSE) --> Disclose source

**Comment by Darin755** [comment]
Yes as long as they are static html images and text that aren't based on search results. You could create a space on the page that can be rented by various companiea
